numpy/_core/src/umath/fast_loop_macros.h

```diff
@@ -90,6 +90,16 @@ abs_ptrdiff(char *a, char *b)
     BINARY_DEFS\
     BINARY_LOOP_SLIDING
 
+/** (ip1, ip2) -> (op1), for case ip2 has zero stride*/
+#define BINARY_DEFS_ZERO_STRIDE                          \
+    char *ip1 = args[0], *ip2 = args[1], *op1 = args[2]; \
+    npy_intp is1 = steps[0], os1 = steps[2];             \
+    npy_intp n = dimensions[0];                          \
+    npy_intp i;
+
+#define BINARY_LOOP_SLIDING_ZERO_STRIDE \
+    for (i = 0; i < n; i++, ip1 += is1, op1 += os1)
```

Comment:
Could you try to delete these two macros again?  The compiler should be smart enough to replace `steps[1]` with `0` in a block that says `if (steps[1] == 0)`.

Reply:
Yes, seems to work! (off course there might be some (older?) compilers that cannot do this optimization)